addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1017,6 +1017,11 @@ const Wysiwyg = Widget.extend({
         if (linkEl && !linkEl.matches(this.customizableLinksSelector)) {
             return;
         }
+        if (options.link && options.link.querySelector(mediaSelector) && !options.link.textContent.trim()) {
+            // If the link contains a media without text, the link is editable
+            // in the media options instead.
+            return;
+        }
```

Comment:
@bso-odoo Tricky stuff this PR :grimacing: The code related to link edition is a mess :grimacing:

I see that the next condition after this code you are adding is "if there is a snippetsMenu", it's ugly but should not your condition first check that too ? if there is no snippet menu, I don't think you should leave immediately in the case you are handling?